config/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package config
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"time"
+)
+
+var (
+	Config = viper.New()
```

Comment:
We could easily isolate ourselves from viper by defining `var Config Handler`. Then extend `Writer` interface with `func ReadInConfig()` and add a function like this:

```
func Init(c Handler) {
Config = c
}
```

Then in the code one can do something like this:
```
import "github.com/spf13/viper"

...
v := viper.New()
v.SetConfigFile(mypath)
config.Init(v)
```

On a side note, one could just use viper. The only thing that the common package adds is validators, but I have not seen it being used anywhere.